tests/testthat/test-textwriter.r

```diff
@@ -0,0 +1,102 @@
+context("textwriter")
+
+describe("TextWriter", {
+  it("basically works", {
+    tw <- TextWriter$new()
+    on.exit(tw$close())
+
+    expect_identical(tw$readAll(), "")
+
+    tw$write("")
+    expect_identical(tw$readAll(), "")
+
+    tw$write("line one")
+    expect_identical(tw$readAll(), "line one")
+
+    tw$write("\nanother line")
+    expect_identical(tw$readAll(), "line one\nanother line")
+
+    tw$write("more content")
+    expect_identical(tw$readAll(), "line one\nanother linemore content")
+  })
+
+  it("bookmarks are respected", {
+    tw <- TextWriter$new()
+    on.exit(tw$close())
+
+    tw$restorePosition()
+    expect_identical(tw$readAll(), "")
+
+    tw$write("foo")
+    tw$restorePosition()
+    expect_identical(tw$readAll(), "")
```

Comment:
I don't understand this one. Wouldn't we expect `foo`? `restorePosition` is a no op since it hasn't been set, so I'm not seeing why we'd get `""`?

Reply:
ohhh marked defaults to the beginning of the string. So we just undid our write.